src/transformers/backbone.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+from abc import abstractmethod
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from torch import nn
+
+
+@dataclass
+class ShapeSpec:
+    """
+    A simple structure that contains basic shape specification about a tensor. It is often used as the auxiliary
+    inputs/outputs of models, to complement the lack of shape inference ability among pytorch modules.
+    """
+
+    channels: Optional[int] = None
+    height: Optional[int] = None
+    width: Optional[int] = None
+    stride: Optional[int] = None
+
+
+class Backbone(nn.Module):
```

Comment:
The Transformers library does not use abstract classes. Especially not for a new API we haven't quite figured out yet. So let's just do backbone models for now.